push/include/prometheus/gateway.h

```diff
@@ -24,7 +28,8 @@ class PROMETHEUS_CPP_PUSH_EXPORT Gateway {
   Gateway(const std::string& host, const std::string& port,
           const std::string& jobname, const Labels& labels = {},
           const std::string& username = {}, const std::string& password = {},
-          std::chrono::seconds timeout = {});
```

Comment:
Up to now we provided an abstraction to curl. But I can see the necessity to set arbitrary options.

Could you please add a second ctor with your presetup hook that does not have the `username`, `password`, and `timeout` parameters? Those could all be set within the new hook.